zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListenerBuilder.java

```diff
@@ -0,0 +1,161 @@
+     *
+     * @param client the curator framework instance.
+     */
+    public ZooKeeperUpdatingListenerBuilder(CuratorFramework client, String node) {
+        this.client = requireNonNull(client, "client");
+        checkArgument(!isNullOrEmpty(node), "node");
+        this.node = node;
+    }
+
+    /**
+     * Creates a {@link ZooKeeperUpdatingListenerBuilder} with a Zookeeper Connection String and a zNode.
+     *
+     * @param connect the zookeeper connection string.
+     * @param node the zookeeper node to register.
+     */
+    public ZooKeeperUpdatingListenerBuilder(String connect, String node) {
+        checkArgument(!isNullOrEmpty(connect), "connect");
+        checkArgument(!isNullOrEmpty(node), "node");
+        this.connect = connect;
+        this.node = node;
+    }
+
+    /**
+     * Sets the connect timeout (in ms). (default: 1000)
+     *
+     * <p>Ignored when {@link CuratorFramework} is set. (see: {@link #client})
+     *
+     * @param timeout the connect timeout.
+     */
+    public ZooKeeperUpdatingListenerBuilder connectTimeout(int timeout) {
+        checkArgument(timeout > 0, "timeout: %s (expected: > 0)", timeout);
+        this.connectTimeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the session timeout (in ms). (default: 10000)
+     *
+     * <p>Ignored when {@link CuratorFramework} is set. (see: {@link #client})
+     *
+     * @param timeout the session timeout.
+     */
+    public ZooKeeperUpdatingListenerBuilder sessionTimeout(int timeout) {
+        checkArgument(timeout > 0, "timeout: %s (expected: > 0)", timeout);
+        this.sessionTimeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the {@link Endpoint} to register.
+     *
+     * @param endpoint the {@link Endpoint} to register.
+     */
+    public ZooKeeperUpdatingListenerBuilder endpoint(Endpoint endpoint) {
+        this.endpoint = requireNonNull(endpoint, "endpoint");
+        return this;
+    }
+
+    /**
+     * Sets the {@link NodeValueCodec} to encode or decode zookeeper data.
+     *
+     * @param nodeValueCodec the {@link NodeValueCodec} instance to use.
+     */
+    public ZooKeeperUpdatingListenerBuilder nodeValueCodec(NodeValueCodec nodeValueCodec) {
+        this.nodeValueCodec = requireNonNull(nodeValueCodec, "requireNonNull");
+        return this;
+    }
+
+    /**
+     * Returns a newly-created {@link ServerListener} instance that registers the server to zookeeper
+     * when the server starts.
+     */
+    public ZooKeeperUpdatingListener build() {
+        if (client == null) {
+            ExponentialBackoffRetry retryPolicy = new ExponentialBackoffRetry(DEFAULT_CONNECT_TIMEOUT, 3);
+            client = CuratorFrameworkFactory.builder()
+                                            .connectString(connect)
+                                            .retryPolicy(retryPolicy)
```

Comment:
Is there any chance that a user wants set up a different `RetryPolicy`?

Reply:
I guess in that case a user can specify `CuratorFramework` he or she created it by self.